### PR TITLE
SDL 2.0.14

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **sdl/2.0.14**

I tried to import the sdl library from https://github.com/bincrafters/community/tree/main/recipes/sdl2/

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by MartinDelille** [comment]
I still have this error that I don't really know how to handle: https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H032

Can I simply mark the case in need of `SystemPackageTool` has a `ConanInvalidConfiguration` ?

**Comment by SpaceIm** [comment]
Few issues:
- conandata.yml is missing
- add 2.0.14 only for the moment, not 2.0.12
- all system_requirements must be abstracted in a system recipe

**Comment by MartinDelille** [comment]
> - conandata.yml is missing

In fact config.yml was missing but it's fixed

> - add 2.0.14 only for the moment, not 2.0.12

done

> - all system_requirements must be abstracted in a CCI recipe

Can you be more explicit about this issue I raised [here](https://github.com/conan-io/conan-center-index/pull/5466#issuecomment-834131227) ?

**Comment by MartinDelille** [comment]
My question is partially answered here: https://github.com/conan-io/conan-center-index/issues/641

**Comment by MartinDelille** [comment]
And what about a partial support for MacOS and Windows ?

File: recipes/sdl2/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+        "vulkan": True
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
```
Comment (SpaceIm) [inline recipes/sdl2/all/conanfile.py]: I would also remove iconv (maybe only for Visual Studio). In this branch https://github.com/libsdl-org/SDL/blob/889cebb7c20d4195e5d4ac344a2175f2490354cc/CMakeLists.txt#L779, iconv is never tested.

**Comment by Croydon** [comment]
> And what about a partial support for MacOS and Windows ?

Theoretically possible. If you ask me personally, I think it will once again confuse people and pretty much doubles maintenance work.

File: recipes/sdl/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+            self._check_pkg_config(self.options.esd, "esound")
+            self._check_pkg_config(self.options.wayland, "wayland-client")
+            self._check_pkg_config(self.options.wayland, "wayland-protocols")
+            self._check_pkg_config(self.options.directfb, "directfb")
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._check_dependencies()
+
+            self._cmake = CMake(self)
```
Comment (prince-chrismc) [inline recipes/sdl/all/conanfile.py:175]:
```suggestion
    def validate(self):
        self._check_dependencies()

    def _configure_cmake(self):
        if not self._cmake:
            self._cmake = CMake(self)
```

File: recipes/sdl/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+        del self.info.options.sdl2main
+
+    def build_requirements(self):
+        if self.settings.os == "Linux":
+            self.build_requires("pkgconf/1.7.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "SDL2-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
```
Comment (prince-chrismc) [inline recipes/sdl/all/conanfile.py:138]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

File: recipes/sdl/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,319 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.29.1"
```
Comment (prince-chrismc) [inline recipes/sdl/all/conanfile.py:5]:
```suggestion
required_conan_version = ">=1.33.0"
```

File: recipes/sdl/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+            elif self.settings.os == "Windows":
+                self._cmake.definitions["DIRECTX"] = self.options.directx
+
+            self._cmake.configure(build_dir=self._build_subfolder)
+        return self._cmake
+
+    def _build_cmake(self):
+        if self.settings.os == "Linux":
+            if self.options.pulse:
+                os.rename("libpulse.pc", "libpulse-simple.pc")
```
Comment (prince-chrismc) [inline recipes/sdl/all/conanfile.py:236]:
```suggestion
                tools.rename("libpulse.pc", "libpulse-simple.pc")
```

**Review by prince-chrismc — COMMENTED** [review]
Minor editorial

**Review by prince-chrismc — APPROVED** [review]
LGTM

File: recipes/sdl/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+            if self.options.esd:
+                self._add_libraries_from_pc("esound")
+            if self.options.directfb:
+                self._add_libraries_from_pc("directfb")
+            if self.options.video_rpi:
+                self.cpp_info.components["libsdl2"].libs.append("bcm_host")
+                self.cpp_info.components["libsdl2"].includedirs.extend([
+                    "/opt/vc/include",
+                    "/opt/vc/include/interface/vcos/pthreads",
+                    "/opt/vc/include/interface/vmcs_host/linux"
```
Comment (uilianries) [inline recipes/sdl/all/conanfile.py:300]:
```suggestion
                    "/opt/vc/include",
                    "/opt/vc/include/interface/vcos/pthreads",
                    "/opt/vc/include/interface/vmcs_host/linux"
```
Why those absolute path, can it be different? You add options and remove them from package ID, so the user can pass any path.

Comment (SSE4) [reply recipes/sdl/all/conanfile.py:300]: they are hard-coded in RPI AFAIK

**Comment by MartinDelille** [comment]
I agree with @jgsogo [comment](https://github.com/conan-io/conan-center-index/pull/5466#issuecomment-835351470) worrying about double maintenance. I'll be interested to follow how to tackle the issue with system requirement.

**Review by jgsogo — DISMISSED** [review]
IMHO, this recipe is good enough (really good job!) as a starting point to provide SDL. I'm sure it will require some iterations and adjustments, but it is starting to work... and some people will benefit from it.

**Review by prince-chrismc — DISMISSED** [review]

**Review by SSE4 — DISMISSED** [review]

File: recipes/sdl/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Macos" and not self.options.iconv:
+            raise ConanInvalidConfiguration("On macOS iconv can't be disabled")
+        if self.settings.os == "Linux":
+            raise ConanInvalidConfiguration("Linux not supported yet")
```
[the remark this answers is not in the document]
Comment (MartinDelille) [reply recipes/sdl/all/conanfile.py]: done

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Comment by SpaceIm** [comment]
Why it is not `sdl2`? I think it's a very bad idea, all package managers use a different name for sdl (1) and sdl2.

**Comment by jgsogo** [comment]
Hi, @SpaceIm (thanks for putting your other PR in draft). I agree with you, I think it's been a mistake and I would vote to rename this recipe (and _deprecate_ current `sdl` one). 

What do other reviewers think?

**Comment by prince-chrismc** [comment]
No objection

**Comment by jgsogo** [comment]
I've created PR here: https://github.com/conan-io/conan-center-index/pull/5944

**Pull request merged**